src/engines/mod.rs

```diff
@@ -578,7 +580,11 @@ fn input_seek<'a, I: 'a + IoProvider>(es: *mut ExecutionState<'a, I>, handle: *m
         libc::SEEK_SET => SeekFrom::Start(offset as u64),
         libc::SEEK_CUR => SeekFrom::Current(offset as i64),
         libc::SEEK_END => SeekFrom::End(offset as i64),
-        _ => panic!("Unexpected \"whence\" parameter to fseek() wrapper: {}", whence),
+        _ => {
+                tt_warning!(es.status, "Unexpected \"whence\" parameter to fseek() wrapper: {}",
+                            whence);
+                return 0;
```

Comment:
Thanks for the suggestion. Unfortunately, as far as can tell, the semantics of `input_seek` are not clear. It is not really a wrapper around `fseek`, so that message is a bit misleading. Looking at the history, it was used to replace various methods, mostly [`rewind`](http://www.cplusplus.com/reference/cstdio/rewind/), which does not report errors. In fact, the code does not check the return value, except at two places in `dpx-pdfobj.c` like [this inefficient looking search for the beginning of the line](https://github.com/rekka/tectonic/blob/829a9647434579642dbd6c7d3962b8c52f20e2b6/tectonic/dpx-pdfobj.c#L2672-L2678). There it means the current position. I agree that it would be definitely better to abort the engine and report a proper error. I think that it will need a more serious refactoring of the bridge code. This commit at least fixes the undefined behavior.